### DISCUSSION TITLE
absent_over_time not working without filters

### DISCUSSION THREAD

**Discussion by rbjorklin** [prometheus/prometheus#9411] [Q&A] [answered]
**What did you do?** Tried using [absent_over_time](https://prometheus.io/docs/prometheus/latest/querying/functions/#absent_over_time) in alert rules.
Expression: `absent_over_time(dell_hw_storage_pdisk_status[5m]) > 0`

**What did you expect to see?** Expression evaluate to true and alerts firing

**What did you see instead? Under which circumstances?** I just get `0 series returned` however if I apply a filter to a known bad node the alert works: `absent_over_time(dell_hw_storage_pdisk_status{node="example1"}[5m]) > 0`

**Environment**

* System information:

```
$ uname -srm
Linux 4.19.0-9-amd64 x86_64
```

* Prometheus version:

```
$ prometheus --version
prometheus, version 2.29.1 (branch: HEAD, revision: dcb07e8eac34b5ea37cd229545000b857f1c1637)
  build user:       root@364730518a4e
  build date:       20210811-14:48:27
  go version:       go1.16.7
  platform:         linux/amd64
```

**Comment by roidelapluie** [comment, marked as answer]
Given how prometheus work, absent over time does not know which labels to expect.

you can use 

```promql
present_over_time(dell_hw_storage_pdisk_status[1h]) unless dell_hw_storage_pdisk_status
```

for a similar result.
